timemachine/cpp/src/nonbonded_all_pairs.cu

```diff
@@ -21,45 +23,57 @@
-                    &k_nonbonded_unified<RealType, 0, 0, 1, 1>,
-                    &k_nonbonded_unified<RealType, 0, 1, 0, 0>,
-                    &k_nonbonded_unified<RealType, 0, 1, 0, 1>,
-                    &k_nonbonded_unified<RealType, 0, 1, 1, 0>,
-                    &k_nonbonded_unified<RealType, 0, 1, 1, 1>,
-                    &k_nonbonded_unified<RealType, 1, 0, 0, 0>,
-                    &k_nonbonded_unified<RealType, 1, 0, 0, 1>,
-                    &k_nonbonded_unified<RealType, 1, 0, 1, 0>,
-                    &k_nonbonded_unified<RealType, 1, 0, 1, 1>,
-                    &k_nonbonded_unified<RealType, 1, 1, 0, 0>,
-                    &k_nonbonded_unified<RealType, 1, 1, 0, 1>,
-                    &k_nonbonded_unified<RealType, 1, 1, 1, 0>,
-                    &k_nonbonded_unified<RealType, 1, 1, 1, 1>}),
+    : N_(lambda_offset_idxs.size()), K_(atom_idxs ? atom_idxs->size() : N_), cutoff_(cutoff), nblist_(K_), beta_(beta),
+      d_atom_idxs_(nullptr), d_sort_storage_(nullptr), d_sort_storage_bytes_(0), nblist_padding_(0.1),
+      disable_hilbert_(false), kernel_ptrs_({// enumerate over every possible kernel combination
+                                             // U: Compute U
+                                             // X: Compute DU_DL
+                                             // L: Compute DU_DX
+                                             // P: Compute DU_DP
+                                             //                             U  X  L  P
+                                             &k_nonbonded_unified<RealType, 0, 0, 0, 0>,
+                                             &k_nonbonded_unified<RealType, 0, 0, 0, 1>,
+                                             &k_nonbonded_unified<RealType, 0, 0, 1, 0>,
+                                             &k_nonbonded_unified<RealType, 0, 0, 1, 1>,
+                                             &k_nonbonded_unified<RealType, 0, 1, 0, 0>,
+                                             &k_nonbonded_unified<RealType, 0, 1, 0, 1>,
+                                             &k_nonbonded_unified<RealType, 0, 1, 1, 0>,
+                                             &k_nonbonded_unified<RealType, 0, 1, 1, 1>,
+                                             &k_nonbonded_unified<RealType, 1, 0, 0, 0>,
+                                             &k_nonbonded_unified<RealType, 1, 0, 0, 1>,
+                                             &k_nonbonded_unified<RealType, 1, 0, 1, 0>,
+                                             &k_nonbonded_unified<RealType, 1, 0, 1, 1>,
+                                             &k_nonbonded_unified<RealType, 1, 1, 0, 0>,
+                                             &k_nonbonded_unified<RealType, 1, 1, 0, 1>,
+                                             &k_nonbonded_unified<RealType, 1, 1, 1, 0>,
+                                             &k_nonbonded_unified<RealType, 1, 1, 1, 1>}),
```

Comment:
side note: these huge diffs from minor changes to the initializer list are pretty annoying; I'm looking forward to eventually upgrading to `clang-format` 14 which has the [PackConstructorInitializers](https://clang.llvm.org/docs/ClangFormatStyleOptions.html#:~:text=define%20BAR%0A%23endif-,packconstructorinitializers,-([PackConstructorInitializers](https://clang.llvm.org/docs/ClangFormatStyleOptions.html#:~:text=define%20BAR%0A%23endif-,packconstructorinitializers,-(PackConstructorInitializersStyle)%20clang-format)Style)%20clang-format) setting to better control this